src/main/java/org/apache/cassandra/sidecar/MainModule.java

```diff
@@ -311,4 +320,28 @@ public ChecksumVerifier checksumVerifier(Vertx vertx)
     {
         return new MD5ChecksumVerifier(vertx.fileSystem());
     }
+
+    @Provides
+    @Singleton
+    @Named("SidecarVersion")
+    public String sidecarVersion()
+    {
+        String resource = "/sidecar.version";
+        try (InputStream input = NodeSettings.class.getResourceAsStream(resource);
+             ByteArrayOutputStream output = new ByteArrayOutputStream())
+        {
+            byte[] buffer = new byte[32];
+            int length;
+            while ((length = input.read(buffer)) >= 0)
+            {
+                output.write(buffer, 0, length);
+            }
+            return output.toString(StandardCharsets.UTF_8.name());
+        }
+        catch (Exception exception)
+        {
+            LOGGER.error("Failed to retrieve Sidecar version from resource {}", resource, exception);
```

Comment:
What's the consequence of not having the correct sidecar version? IMO, this should be fatal. If sidecar version is expected and is not available, we should bail on the start.

@frankgh @yifan-c any opinions?

Reply:
To explain my motivation: if we don't have a resource with version build into Sidecar, then something went wrong with the artifact build process, but it doesn't prevent us from starting up and serving requests, so just log an error, and return "`unknown`", which is only used for usage dashboard in Splunk.

Reply:
Well, we will definitely not have the version for development, unit testing, or some gradle targets. Which means that we will need to handle those cases if we are going to go with that behavior.

As Yuriy mentioned above, if the file is missing it means there was some issue during the build process. I think we should not worry much about that case. I'm leaning more towards not failing the startup for two reasons. The first reason is that we should/can rely on the build process to correctly generate the version file. The second reason is that we will need to handle the cases where it's fine to not have the version (i.e unit tests, gradlew run, etc)